onnxruntime/contrib_ops/cuda/layer_norm_impl.cu

```diff
@@ -355,19 +356,66 @@ void HostApplyLayerNorm(
   const uint64_t maxGridY = prop.maxGridSize[1];
   const int warp_size = prop.warpSize;
   ORT_ENFORCE(warp_size == GPU_WARP_SIZE);
+  
+  bool custom_supported_dim = std::find(std::begin(SUPPORTED_REDUCTION_DIM), std::end(SUPPORTED_REDUCTION_DIM), n2) != std::end(SUPPORTED_REDUCTION_DIM);
+  bool supported_half2 = (prop.major >= 7) || !(std::is_same<T, half>::value);
```

Comment:
half2 is supported from 6.0

Reply:
The DeepSpeed kernel uses a min version 7.0 in their code, so this is replicated here. On digging, I found that in compute capability 6.x, _half2 is supported but at a lower rate and might not have perf gain.
https://forums.developer.nvidia.com/t/half2-performance/66652/3
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#arithmetic-instructions
 https://forums.developer.nvidia.com/t/fp16-support-on-gtx-1060-and-1080/53256

Reply:
No, from computation point of view, half2 is 2x of half on GPUs from Pascal generation.